src/modules/user/user.service.ts

```diff
@@ -1,37 +1,51 @@
 import { Injectable } from '@nestjs/common';
 import { InjectRepository } from '@nestjs/typeorm';
 import { Repository } from 'typeorm';
+import { slugify } from 'transliteration';
 import { User } from '../database/entities/user.entity';
 import { CreateUserDto } from './dto/received/create-user.dto';
 import { MessageCodeError } from '../../shared/errors/message-code-error';
 import { UpdateUserDto } from './dto/received/update-user.dto';
+import { SocialLink } from '../database/entities/socialLinks.entity';
 
 @Injectable()
 export class UserService {
   constructor(
     @InjectRepository(User)
     private readonly userRepository: Repository<User>,
+    @InjectRepository(SocialLink)
+    private readonly socialLinkRepository: Repository<SocialLink>,
   ) {}
 
   async getUsers(): Promise<User[]> {
-    return this.userRepository.find();
+    return this.userRepository.find({ relations: ['socialLinks'] });
   }
 
   async getUserById(id: string): Promise<User> {
-    return this.userRepository.findOne(id);
+    return this.userRepository.findOne(id, { relations: ['socialLinks'] });
   }
 
   async createUser(data: CreateUserDto): Promise<User> {
-    const existUser = await this.userRepository.findOne({ email: data.email });
+    const existUser = await this.userRepository.findOne({ name: data.name });
     if (existUser) throw new MessageCodeError('user:exist');
+
     const newUser = this.userRepository.create(data);
-    await this.userRepository.save(data);
-    return newUser;
+    newUser.seoId = slugify(data.name);
```

Comment:
seoId unique field, await this.userRepository.findOne({ name: data.name }); user or condition here to find by one of fields

Reply:
done